From 256b62df6654305b3ceebb14089b62b9996d5f08 Mon Sep 17 00:00:00 2001
From: Richard W.M. Jones <rjones@redhat.com>
Date: Mon, 28 Apr 2014 09:14:34 +0100
Subject: [PATCH 21/67] Add a regression test for RHBZ#1091803.

(cherry picked from commit 50b207ca459a205efeec613ab04ff4c39dc6bee2)
---
 tests/regressions/Makefile.am    |    1 +
 tests/regressions/rhbz1091803.sh |   39 ++++++++++++++++++++++++++++++++++++++
 2 files changed, 40 insertions(+), 0 deletions(-)
 create mode 100755 tests/regressions/rhbz1091803.sh

diff --git a/tests/regressions/Makefile.am b/tests/regressions/Makefile.am
index 358e8d0..f0d60cd 100644
--- a/tests/regressions/Makefile.am
+++ b/tests/regressions/Makefile.am
@@ -33,6 +33,7 @@ TESTS = \
 	rhbz957772.sh \
 	rhbz975797.sh \
 	rhbz1001875.sh \
+	rhbz1091803.sh \
 	test-noexec-stack.pl
 
 tests_not_run = \
diff --git a/tests/regressions/rhbz1091803.sh b/tests/regressions/rhbz1091803.sh
new file mode 100755
index 0000000..e9be95d
--- /dev/null
+++ b/tests/regressions/rhbz1091803.sh
@@ -0,0 +1,39 @@
+#!/bin/bash -
+# libguestfs
+# Copyright (C) 2014 Red Hat Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+# Regression test for:
+# https://bugzilla.redhat.com/show_bug.cgi?id=1091803
+# tar-in API didn't cancel the receive correctly along all error paths.
+
+set -e
+export LANG=C
+
+if [ -n "$SKIP_TEST_RHBZ1091803_SH" ]; then
+    echo "$0: test skipped because environment variable is set."
+    exit 77
+fi
+
+../../fish/guestfish <<EOF
+scratch 100M
+run
+mkfs ext2 /dev/sda
+mount /dev/sda /
+-tar-in nosuchtarfile.tar /nosuchdirectory
+# Appliance has now crashed, so any subsequent command fails:
+ping-daemon
+EOF
-- 
1.7.1

